src/libs/actions/IOU.js

```diff
@@ -111,8 +111,32 @@ function createIOUSplit(params) {
         });
 }
 
+/**
+ * Settles an IOU Report. As we can guarantee that the settled report was previously active,
+ * we should update the chatReport data too when calling `fetchIOUReportByID`.
```

Comment:
As mentioned in the longer explanation, we currently depend on both ChatReport **and**  IOUReport in order to display IOUs. We maintain a link between these reports locally: 

```
ChatReport: {
    id: 9876,
    iouReportID: 1234,
    hasOutstandingIOU: true,
}

IOUReport: {
    id: 1234,
    chatReportID: 9876,
}
```

If we settle a report but only update the IOU report, then that chat report will still be associated with an old settled report. Technically the incoming push notification should also trigger an update to the report, but it seems better to immediately update the report.

Reply:
> What does "previously active" mean in this context?

By 'Previously active' I meant that the IOU was still 'open'. 

The comment is trying to explain why we pass true as the third argument on line 134 (I couldn't find a good way to add this long comment inline). SO, if I refactor the function to two individual functions as you suggest, this will not need to be stated.

Reply:
Got it, so I guess my follow up here would be... 

Why wouldn't we always want to update the chat report and the iou report at the same time?

>By 'Previously active' I meant that the IOU was still 'open'.

Got it thanks! Can just say "IOU open" or "outstanding balance" or use whatever language we are using elsewhere for consistency?

Reply:
>If the user is viewing an old, settled chat report while another newer IOU report is open, then we must be careful to not update the chatReport -- because doing so would overwrite the chatReports iouReportID prop. This would mean that the chatReport now points to the old settled iouReport instead of the open iouReport.

Hmm, can you maybe explain this a different way? I'm still not quite getting it. It sounds like:

1. A chat report can be "old" and "settled". Are those states the same?
2. An IOU report can be "open". Does that mean we are viewing it somewhere or that it's in an "open" state?

>This would mean that the chatReport now points to the old settled iouReport instead of the open iouReport

3. Why is this a problem? What exactly would the user experience?

Reply:
> Got it thanks! Can just say "IOU open" or "outstanding balance" or use whatever language we are using elsewhere for consistency?

 I don't think we've settled on a term yet -- which certainly doesn't help us here 😅 , but IOU Open is probably the closest we have settled on 👍

_____

> An IOU report can be "open". Does that mean we are viewing it somewhere or that it's in an "open" state?

This just means an open state. Any IOUReport can be viewed at any time, no matter whether it is 'open' or 'settled|closed' 

_____

> A chat report can be "old" and "settled". Are those states the same?

Apologies, it was late when I wrote that yesterday and I made a mistake, it should have said: 'If the user is viewing an old, settled _**IOU report**_ while another newer IOU report is open...'

Only IOUReports have 'states' that are relevant to our discussion:
- A) First IOU is created: -> IOUReport_1 is created, transaction added
- B) Another IOU is created -> the transaction is added to IOUReport_1
- C) IOU settled -> IOUReport_1 is settled (closed?)
- D) Another IOU is created -> IOUReport_2 is created, transaction added

So, an IOUReport_1 is in an 'open' state from the time it is created (A), to the time it is settled (C). While an IOUReport is 'open', the chatReport must know this IOUReport ID, so that we know whether to show the 'hasOutstandingIOU' Badge and Preview components or not.

So to answer your previous question again:
> Why wouldn't we always want to update the chat report and the iou report at the same time?

1) Settling a report -> The chatReport _must_ be updated to remove the chatReport association, as there is no longer an open IOU
2) User views an open IOU -> There is _no harm_ in updating the chatReport in this case, because it will still point to the open IOU
3) User views an old, settled IOU (while a newer IOU is open) -> This is the problematic case. If we update the chatReport after retrieving this IOUReport with our current logic, then the chatReport would be wrongly pointed to the old chatReport, overwriting the ID of the open chat report

This is where the `shouldUpdateChatReport` flag came in, I wanted to allow for case 3. Since your review though, I have separated this into two functions: 
- `fetchIOUReportByID` for case 3
- `fetchIOUReportByIDAndUpdateChatReport` for case 1 & 2

Reply:
Ok, tbh I'm still a little confused. It seems like we are working very hard to manage the state of the chat report in the front end depending on the situation. And the situation you are describing is not very obvious from looking at the code. Maybe we just need to add some more specific comments that explain the possibility and why it's a concern?

Reply:
Fair enough -- I agree that this tricky, hidden logic is far from ideal. 

I've tried to simplify this by documenting the [two functions here](https://github.com/Expensify/Expensify.cash/blob/50f8dbbd041982d0dfbe1449a4aa9d1489a8788b/src/libs/actions/Report.js#L447-L471), and in every place that they are called, to explain why one function was chosen over the other.